### PR TITLE
Feature/entry field support in rte

### PR DESCRIPTION
**Pull request by KANE-99**
This will enable two things:

1. RTELocation will have access to latest entry data. 
2. Apps can now pass hoveringToolbarOptions in which they can set autoWidth property, So that app can decide on the icon width. (required for Icons like AI Assistant). This will not affect other plugins icon on the hovering toolbar, we are just allowing a particular app to have width according to its icon size.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -111,6 +111,12 @@ export declare interface ISidebarInitData extends ICommonInitData {
     type: LocationType.WIDGET;
 }
 
+export declare interface IRTELocationInitData extends ICommonInitData {
+    content_type: ContentType;
+    entry: Entry;
+    locale: string;
+}
+
```
Comment (Kaushty) [inline src/types.ts:119]: We already have `IRTEInitData` type definition for RTE Plugins, we could add entry to the same and make use of that instead.

**Review by surajcontentstack — APPROVED** [review]

File: src/types.ts (hunk 1 above)
Comment (KANE-99) [reply src/types.ts:119]: @Nilesh-y can you check this?

**Review by abhishek305 — CHANGES_REQUESTED** [review]
Minor change for type rest looks good !!

File: src/types.ts (hunk 1 above)
Comment (abhishek305) [reply src/types.ts]: @Nilesh-y Could you please check this comment ?

Comment (Nilesh-y) [reply src/types.ts]: Yes @abhishek305 @KANE-99 made the changes.... please check

Comment (abhishek305) [reply src/types.ts]: got it @Nilesh-y

**Review by abhishek305 — APPROVED** [review]

**Pull request merged**